### PR TITLE
Deprecate kickstart modularity module

### PR DESCRIPTION
**Pull request by marusak**
@jkonecny12 You mentioned I should add release-note? Can you please give me some hint?

### CODE REVIEW

**Review by jkonecny12 — CHANGES_REQUESTED** [review]
Please add also release notes so users are aware that `module` command is deprecated on next major release.

**Comment by marusak** [comment]
@jkonecny12 added release note

File: docs/release-notes/deprecate-modularity.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+:Type: Kickstart
+:Summary: Deprecate RPM modularity module
+
+:Description:
+    Based on the DNF team recommendation we decided to deprecate RPM modularity
+    feature in Anaconda. This is only a deprecation not removal and we would
+    evaluate RHEL-11 status when it's near based on decisions from the DNF team.
```
Comment (KKoukiou) [inline docs/release-notes/deprecate-modularity.rst:7]: These are release notes for Fedora and not used for RHEL as far as i know. I think it's correct to remove the RHEL mention, and add a Fedora equivalent if applicable.

Comment (marusak) [reply docs/release-notes/deprecate-modularity.rst]: Amended

**Comment by jkonecny12** [comment]
The validate test is expected to fail because your change won't force container rebuild. Let's merge this to get a new containers.

**Review by jkonecny12 — APPROVED** [review]

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**